go/db/db.go

```diff
@@ -113,7 +117,13 @@ func openOrchestratorMySQLGeneric() (db *sql.DB, fromCache bool, err error) {
 	if config.Config.MySQLOrchestratorUseMutualTLS {
 		uri, _ = SetupMySQLOrchestratorTLS(uri)
 	}
-	return sqlutils.GetDB(uri)
+	if db, fromCache, err = sqlutils.GetDB(uri); err != nil {
+		return nil, fromCache, err
+	}
+	if config.Config.MySQLConnectionLifetime > 0 {
+		db.SetConnMaxLifetime(time.Duration(config.Config.MySQLConnectionLifetime) * time.Second)
+	}
```

Comment:
This is not needed. If the `db` is returned from cache then `SetConnMaxLifetime` has already been applied.

Reply:
Please remove as per comment.

Reply:
Reverted to the original code here, as per your request